app/scripts/controllers/transactions/lib/util.js

```diff
@@ -218,12 +220,37 @@ export function validateTxParams(txParams, eip1559Compatibility = true) {
           );
         }
         break;
+      case 'data':
+        validateInputData(value);
```

Comment:
`ensureFieldIsString(txParams, 'data');` needs to be added here, below line 224

This is done for the cases above, and will ensure that tests pass.

Reply:
Added in https://github.com/MetaMask/metamask-extension/pull/17326/commits/4068f514146a1b5e93325ce157d13f215483b4a4